### PR TITLE
Aryan todos story test

### PR DESCRIPTION
**Pull request by Aryan-Akhilesh**
Added all the story tests

### CODE REVIEW

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+public class TodosStepDefinition {
+
+    private final String json = "application/json";
+    private final String xml = "application/xml";
+    private Response res;
+    private String jsonString;
+    private String xmlString;
```
Comment (Lobo808) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:17]: Remove useless field

**Review by Lobo808 — CHANGES_REQUESTED** [review]
Just add more examples to it, using different IDs and string values

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+    // ----------- Feature: Get specific todos -------------//
+
+    @Given("I have an existing todo")
+    public void i_have_an_existing_todo() {
+        // 2 todos already exist
+    }
+
+    @When("I request the todo with a specific id {int} in JSON")
+    public void i_request_the_todo_with_a_specific_id_in_json(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:27]: Is there a reason why you are using Big Int for id?

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    @Then("I should see the todo in JSON")
+    public void i_should_see_the_todo_in_json() {
+        int statusCode = res.getStatusCode();
+        String body = "{\"todos\":[{\"id\":\"1\",\"title\":\"scan paperwork\",\"doneStatus\":\"false\",\"description\":\"\",\"categories\":[{\"id\":\"1\"}],\"tasksof\":[{\"id\":\"1\"}]}]}";
+        JSONAssert.assertEquals(body,res.getBody().asString(), false);
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @When("I request the todo with a specific id {int} in XML")
+    public void i_request_the_todo_with_a_specific_id_in_xml(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:40]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @Given("I have a non-existing todo")
+    public void i_have_a_non_existing_todo() {
+        //nothing to do here
+    }
+
+    @When("I request the todo with an invalid id {int} in JSON")
+    public void i_request_the_todo_with_an_invalid_id_in_json(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:58]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+    @Then("I should see taskOf the todo in XML")
+    public void i_should_see_taskOf_the_todos_in_xml() {
+        int statusCode = res.getStatusCode();
+        Assertions.assertTrue((res.getBody().asString()).matches("<projects>(?:<project>(?:(?!<\\/project>)(?:<active>false<\\/active>|<description\\/>|<id>1<\\/id>|<completed>false<\\/completed>|<title>Office Work<\\/title>|<tasks>(?:(?!<\\/tasks>)(?:<id>2<\\/id>|<id>1<\\/id>))<\\/tasks>)*?)<\\/project>)*<\\/projects>"));
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @When("I get the taskOf the todo with an invalid id {int}")
+    public void i_get_the_task_of_the_todo_with_an_invalid_id(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:235]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        JSONAssert.assertEquals(jsonString,res.getBody().asString(), false);
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @Given("I have existing todos")
+    public void i_have_existing_todos() {
+    }
+
+    @When("I get the taskOf the todo with the valid id {int} in XML")
+    public void i_get_the_task_of_the_todo_with_the_valid_id_in_xml(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:223]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    //-------------- Feature: Get TasksOf todos -------------//
+
+    @Given("I have existing todo")
+    public void i_have_existing_todo() {
+    }
+
+    @When("I get the taskOf the todo with the valid id {int} in JSON")
+    public void i_get_the_task_of_the_todo_with_the_valid_id_in_json(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:206]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    public void i_post_a_todo_with_valid_id_and_different_description(Integer todoID, String description) {
+        jsonString =  "{" + "\"id\": "+todoID+ "," + "\"description\": \""+ description + "\"}";
+        res = RestAssured.given()
+                .header("Content-Type",json)
+                .body(jsonString)
+                .post("http://localhost:4567/todos/" + todoID);
+    }
+
+    @When("I PUT a todo with invalid id {int} and different description {string}")
+    public void i_put_a_todo_with_invalid_id_and_different_description(Integer todoID, String description) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:185]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+        //nothing to do here
+    }
+
+    @When("I request the todo with an invalid id {int} in JSON")
+    public void i_request_the_todo_with_an_invalid_id_in_json(Integer todoID) {
+        res = RestAssured.given().header("Accept",json).get("http://localhost:4567/todos/" + todoID);
+    }
+
+    @Then("I should see no todo with the id {int}")
+    public void i_should_see_no_todo(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:63]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+                .post("http://localhost:4567/todos");
+
+        int statusCode = res.getStatusCode();
+        Assertions.assertEquals(400,statusCode);
+    }
+
+    //------------ Feature: Delete todos -----------//
+
+    @When("I delete the todo with the valid id {int}")
+    public void i_delete_the_todo_with_the_valid_id(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:118]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    @Then("I should not see the todo listed under todos anymore")
+    public void i_should_not_see_the_todo_listed_under_todos_anymore() {
+        int statusCode = res.getStatusCode();
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @When("I post todo with valid id {int} but with the doneStatus {string} as completed")
+    public void i_post_todo_with_valid_id_but_with_the_done_status_as_completed(Integer todoID, String doneStatus) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:129]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+    @Then("I should see the todo but with completed status in todos")
+    public void i_should_see_the_todo_but_with_completed_status_in_todos() {
+        int statusCode = res.getStatusCode();
+        Assertions.assertEquals(200,statusCode);
+        System.out.println(res.getBody().asString());
+    }
+
+    @When("I delete the todo with the invalid id {int}")
+    public void i_delete_the_todo_with_the_invalid_id(Integer todoID) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:145]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+    @Then("The todo is still listed under todos")
+    public void the_todo_is_still_listed_under_todos() {
+        int statusCode = res.getStatusCode();
+        Assertions.assertEquals(404,statusCode);
+    }
+
+    //--------------Feature: Amend Description --------------//
+
+    @When("I PUT a todo with valid id {int} and different description {string}")
+    public void i_put_a_todo_with_valid_id_and_different_description(Integer todoID, String description) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:158]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    @Then("I should see that the description has changed for that todo")
+    public void i_should_see_that_the_description_has_changed_for_that_todo() {
+        int statusCode = res.getStatusCode();
+        Assertions.assertEquals(200,statusCode);
+    }
+
+    @When("I POST a todo with valid id {int} and different description {string}")
+    public void i_post_a_todo_with_valid_id_and_different_description(Integer todoID, String description) {
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java:176]: ditto

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 1 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Removed xmlString

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 2 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 3 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 4 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 5 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 6 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 7 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 8 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 9 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 10 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 11 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 12 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 13 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

File: TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java (hunk 14 above)
Comment (Aryan-Akhilesh) [reply TodoListManagerTest/src/test/java/StoryTest/Todos/TodosStepDefinition.java]: Changed

**Comment by Aryan-Akhilesh** [comment]
Added more examples

**Comment by MosinLover** [comment]
lgtm

**Pull request merged**